airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -152,8 +152,19 @@ private static boolean isCheckWithCatalog(IntegrationConfig parsed) {
 
   private AirbyteMessage runCheckWithCatalog(final IntegrationConfig parsed) throws Exception {
     LOGGER.info("Using Check With Catalog Method");
-    // TODO: Actually implement something different
-    return runCheck(parsed);
+    if (!(integration instanceof Check)) {
+      LOGGER.warn("Catalog Sent to check which does not implement checkWithCatalog for connector {}", integration);
+      return runCheck(parsed);
+    } else {
+      try {
+        final JsonNode config = getValidatedConfig(parsed, CHECK.toString());
+        final ConfiguredAirbyteCatalog catalog = parseConfig(parsed.getCatalogPath(), ConfiguredAirbyteCatalog.class);
+        return new AirbyteMessage().withType(Type.CONNECTION_STATUS).withConnectionStatus(((Check) integration).check(config, catalog));
```

Comment:
I would extract the ((Check) integration).check(config, catalog) for readability

Reply:
done